### PR TITLE
tests: boards: update to new ztest APIs.

### PR DESCRIPTION
**Pull request by aaronemassey**

### CODE REVIEW

File: tests/boards/native_posix/cpu_wait/src/main.c [hunk 1]
```diff
@@ -211,7 +211,7 @@ static void np_timer_isr_test_replacement(const void *arg)
  * The kernel is configured as NOT-tickless, and the default tick period is
  * 10ms
  */
-static void test_cpu_hold_with_interrupts(void)
+ZTEST(native_cpu_hold, test_cpu_hold_with_interrupts)
 {
 #if defined(CONFIG_BOARD_NATIVE_POSIX)
```
Comment (yperess) [inline tests/boards/native_posix/cpu_wait/src/main.c]: This test is a no-op if `!CONFIG_BOARD_NATIVE_POSIX`. Can we move the `ifdef` up so it's just never declared in that case?

**Review by yperess — CHANGES_REQUESTED** [review]

File: tests/boards/native_posix/cpu_wait/src/main.c (hunk 1 above)
Comment (aaronemassey) [reply tests/boards/native_posix/cpu_wait/src/main.c]: Done.

**Comment by aaronemassey** [comment]
Closing PR to make it directly to Zephyr

**Comment by aaronemassey** [comment]
Closing PR to make it directly to Zephyr

**Comment by aaronemassey** [comment]
Closing PR to make it directly to Zephyr

**Pull request closed without merging**